src/shims/x86/sse42.rs

```diff
@@ -459,9 +460,8 @@ pub(super) trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // Perform polynomial division modulo 2.
                 let mut dividend = (crc << bit_size) ^ (v << 32);
                 const POLYNOMIAL: u128 = 0x11EDC6F41;
-                let leading_zeros = dividend.leading_zeros();
-                while leading_zeros <= POLYNOMIAL.leading_zeros() {
-                    dividend ^= (POLYNOMIAL << POLYNOMIAL.leading_zeros()) >> leading_zeros;
+                while dividend.leading_zeros() <= POLYNOMIAL.leading_zeros() {
+                    dividend ^= (POLYNOMIAL << POLYNOMIAL.leading_zeros()) >> dividend.leading_zeros();
```

Comment:
This diff is a functional change, right? Previously `leading_zeros` was a constant, not it is recomputed based on the current `dividend` each round.

Which one is right? Do the tests catch the wrong one?

Reply:
Yes. I tried to remove one call to `leading_zeros`, but the transformation was incorrect. I had to revert it.